src/contrib/actions.rs

```diff
@@ -25,3 +26,23 @@ pub fn create_or_switch_to_workspace(
         wm.focus_workspace(&sel);
     })
 }
+
+/**
+ * Focus a Client with the given class as WM_CLASS or spawn the program with
+ * the given command if no such Client exists. This is useful for key bindings 
+ * that are based on the program you want to work with rather than having to 
+ * remember where things are running.
+ */
+pub fn focus_or_spawn(class: String, command: String) -> FireAndForget {
+    Box::new(move |wm: &mut WindowManager| {
+        let cond = |c: &Client| c.class() == &class;
+        let sel = Selector::Condition(&cond);
+        let client = wm.client(&sel);
+        if let Some(client) = client {
```

Comment:
I tend to prefer inlining things like this in an `if let` if possible:
```suggestion
        if let Some(client) = wm.client(&sel) {
```

Reply:
I agree, that does look better. I had more things inlined at first, but the borrow checker didn't like it, so I ended up splitting it into multiple lines. That's how it ended up like this.